Zend/zend_operators.c

```diff
@@ -970,13 +970,13 @@ static zend_never_inline void ZEND_FASTCALL add_function_array(zval *result, zva
 		return;
 	}
 	zval tmp;
-	if (result != op1) {
-		ZVAL_ARR(&tmp, zend_array_dup(Z_ARR_P(op1)));
-	} else {
-		ZVAL_COPY_VALUE(&tmp, result);
-		SEPARATE_ARRAY(&tmp);
-	}
+	ZVAL_ARR(&tmp, zend_array_dup(Z_ARR_P(op1)));
```

Comment:
If  ``Z_RECOUNT_P(op1) == 1`` the duplication is completely useless.

Reply:
Unfortunately not, since the array might be behind a reference with RCN, we just don't know from this point.